bdk 0.17

bonomat opened the pull request:
Should fix #1821

If we want to ensure that more addresses are cached, then we need to call it explicitly:  https://github.com/bitcoindevkit/bdk/commit/45767fcaf7c65d5020f99fdc35c147acd6e8d037

dependabot[bot] committed:
Bump bdk from 0.16.1 to 0.17.0

Before sync ensure that we cached 1000 addresses.
If we want to ensure more addresses to be cached, then `wallet.ensure_addresses_cached` is to be used according to the commit in here: https://github.com/bitcoindevkit/bdk/commit/45767fcaf7c65d5020f99fdc35c147acd6e8d037

Bumps [bdk](https://github.com/bitcoindevkit/bdk) from 0.16.1 to 0.17.0.
- [Release notes](https://github.com/bitcoindevkit/bdk/releases)
- [Changelog](https://github.com/bitcoindevkit/bdk/blob/v0.17.0/CHANGELOG.md)
- [Commits](https://github.com/bitcoindevkit/bdk/compare/v0.16.1...v0.17.0)

---
updated-dependencies:
- dependency-name: bdk
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

klochowicz commented on daemon/src/wallet.rs:
```diff
@@ -135,6 +135,8 @@ impl Actor<ElectrumBlockchain> {
 
 impl Actor<ElectrumBlockchain> {
     fn sync_internal(&mut self) -> Result<WalletInfo> {
+        self.wallet.ensure_addresses_cached(1000)?;
```
perhaps it would be worthwhile to expose this as an optional CLI parameter? In case we run into trouble with calculating wallet balance, we would be able to restart with a little tweak to the parameters instead of cutting a new release

bonomat replied:
I dont' think this is necessary. I'd rather make this dynamic and increase slowly until this maximum is reached.

klochowicz reviewed (approved):
great find!

luckysori reviewed (approved):
Looking at the `bdk` code we thought this wouldn't work for some reason, but I trust your judgement. And obviously the PR you link to indicates that it should. Thanks for looking into this!

da-kami reviewed (changes requested):
Thanks for having another go at this!
I'd say we should only merge this after we successfully tested that the testnet setup's `taker_master` wallet balance was picked up correctly. I think it should not be too hard to test that locally. Might be a bit annoying, but I really want to avoid that we run into a problem where user's balances are `0` 😬

da-kami reviewed (approved):
might still want to wait until the release is out to avoid potential problems related to the upgrade (not necessarily just the wallet)

thomaseizinger reviewed (approved):
Nice! Missed that in my original commit.

Would you mind squashing so we don't have a broken commit in the history?

bonomat committed:
Upgrade bdk to 0.18 and bump maia

This commit bumps maia to the latest commit which includes bdk 0.18 as otherwise this would not compile.
